From e6757604466ee03887c9bcf4227545a6373a040d Mon Sep 17 00:00:00 2001
From: Andrey Shvetsov <andrey.shvetsov@k2l.de>
Date: Thu, 8 Feb 2018 16:13:20 +0100
Subject: [PATCH] add dts for MediaLB on M3ULCB KF

This adds the Device Tree for the MediaLB on the M3ULCB Kingfisher board
on R8A7796 SoC

Signed-off-by: Andrey Shvetsov <andrey.shvetsov@k2l.de>
---
 arch/arm64/boot/dts/renesas/Makefile               |  2 +
 .../boot/dts/renesas/r8a7796-m3ulcb-kf-mlp.dts     | 60 ++++++++++++++++++++++
 2 files changed, 62 insertions(+)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a7796-m3ulcb-kf-mlp.dts

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index 32fb4d926dab..502fc4f396d9 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -4,5 +4,7 @@ dtb-$(CONFIG_ARCH_R8A7795) += r8a7795-es1-salvator-x.dtb r8a7795-es1-h3ulcb.dtb
 dtb-$(CONFIG_ARCH_R8A7796) += r8a7796-salvator-x.dtb r8a7796-m3ulcb.dtb
 dtb-$(CONFIG_ARCH_R8A7796) += r8a7796-salvator-xs.dtb
 
+dtb-$(CONFIG_ARCH_R8A7796) += r8a7796-m3ulcb-kf-mlp.dtb
+
 always		:= $(dtb-y)
 clean-files	:= *.dtb
diff --git a/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb-kf-mlp.dts b/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb-kf-mlp.dts
new file mode 100644
index 000000000000..6e962dc498ac
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb-kf-mlp.dts
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the MediaLB and INIC reset support on the M3ULCB
+ * Kingfisher board on r8a7796
+ *
+ * Copyright (C) 2018 K2L GmbH & Co. KG
+ */
+
+#include "r8a7796-m3ulcb.dts"
+
+/ {
+	model = "Renesas M3ULCB Kingfisher board with MediaLB support";
+
+	/delete-node/sound;
+};
+
+&pfc {
+	mlp_pins: mlp {
+		groups = "mlb_3pin";
+		function = "mlb_3pin";
+	};
+};
+
+&gpio4 {
+	most_rst {
+		gpio-hog;
+		gpios = <14 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "MOST RST";
+	};
+};
+
+&i2c4 {
+	i2cswitch4: pca9548@74 {
+		compatible = "nxp,pca9548";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x71>;
+		reset-gpios= <&gpio3 15 GPIO_ACTIVE_LOW>;
+
+		i2c@3 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <3>;
+			/* MOST node(s) */
+		};
+	};
+};
+
+&mlp {
+	pinctrl-0 = <&mlp_pins>;
+	pinctrl-names = "default";
+	interrupts =
+		<GIC_SPI 384 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 385 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 386 IRQ_TYPE_LEVEL_HIGH>;
+	microchip,clock-speed = "1024fs";
+	status = "okay";
+};
+
-- 
2.14.2

